cpg_workflows/stages/aip.py

```diff
@@ -0,0 +1,294 @@
+        job.command(
+            f'python3 reanalysis/query_panelapp.py '
+            f'--panels {str(hpo_panel_json)} '
+            f'--out_path {str(expected_out["panel_data"])}'
+        )
+
+        return self.make_outputs(dataset, data=expected_out, jobs=job)
+
+
+def query_for_latest_mt(dataset: str) -> str:
+    """
+    query for the latest MT for a dataset
+    Args:
+        dataset (str):
+    Returns:
+        str, the path to the latest MT
+    """
+    my_query = gql(
+        """
+        query MyQuery($dataset: String!) {
+            project(name: $dataset) {
+                analyses(type: {eq: "custom"}, status: {eq: COMPLETED}) {
+                    output
+                }
+            }
+        }
+    """
+    )
+    result = query(my_query, variables={'dataset': dataset})
+    seq_type_exome = get_config()['workflow'].get('sequencing_type') == 'exome'
+    mt_path = ''
+    for analysis in result['project']['analyses']:
+        if analysis['output'].endswith('.mt') and (
+            (seq_type_exome and '/exome/' in analysis['output'])
+            or (not seq_type_exome and '/exome/' not in analysis['output'])
+        ):
+            mt_path = analysis['output']
+
+    if not mt_path:
+        raise ValueError(f'No MT found for dataset {dataset}')
+    return mt_path
+
+
+@stage(required_stages=[QueryPanelapp])
+class RunHailFiltering(DatasetStage):
+    """
+    hail job to filter & label the MT
+    """
+
+    def expected_outputs(self, dataset: Dataset) -> dict[str, Path]:
+        return {
+            'labelled_vcf': dataset.prefix() / DATED_FOLDER / 'hail_labelled.vcf.bgz',
+            'pedigree': dataset.prefix() / DATED_FOLDER / 'pedigree.ped',
+        }
+
+    def queue_jobs(self, dataset: Dataset, inputs: StageInput) -> StageOutput:
+
+        # either do as you're told, or find the latest
+        # this could potentially follow on from the AnnotateDataset stage
+        # if this becomes integrated in the main pipeline
+        input_mt = get_config()['workflow'].get(
+            'matrix_table', query_for_latest_mt(dataset.name)
+        )
+        job = get_batch().new_job('run hail labelling')
+        job.image(image_path('aip'))
+        HIGHMEM.set_resources(job, mem_gb=16.0)
+
+        # auth and copy env
+        authenticate_cloud_credentials_in_job(job)
+        copy_common_env(job)
+
+        panelapp_json = inputs.as_str(
+            target=dataset, stage=QueryPanelapp, key='panel_data'
+        )
+        expected_out = self.expected_outputs(dataset)
+        pedigree = dataset.write_ped_file(out_path=expected_out['pedigree'])
+        # peddy can't read cloud paths
+        local_ped = get_batch().read_input(str(pedigree))
+        job.command(
+            f'python3 reanalysis/hail_filter_and_label.py '
+            f'--mt {input_mt} '
+            f'--panelapp {panelapp_json} '
+            f'--pedigree {local_ped} '
+            f'--vcf_out {str(expected_out["labelled_vcf"])}'
```

Comment:
Would recommend quoting these to stop a shell injection attack, or an accidental space.